src/shared/components/cancerSummary/SummaryBarGraph.tsx

```diff
@@ -0,0 +1,215 @@
+        // Tooltip Element
+        let tooltipEl = document.getElementById('chartjs-tooltip');
+
+        // Create element on first render
+        if (!tooltipEl) {
+            tooltipEl = document.createElement('div');
+            tooltipEl.id = 'chartjs-tooltip';
+            tooltipEl.innerHTML = "<div></div>";
+            document.body.appendChild(tooltipEl);
+        }
+
+        // Hide if no tooltip
+        if (tooltipModel.opacity === 0) {
+            tooltipEl.style.opacity = '0';
+            // tooltipEl.style.display = 'none';
+            return;
+        }
+
+        // Set caret Position
+        tooltipEl.classList.remove('above', 'below', 'no-transform');
+        if (tooltipModel.yAlign) {
+            tooltipEl.classList.add(tooltipModel.yAlign);
+        } else {
+            tooltipEl.classList.add('no-transform');
+        }
+
+        function getBody(bodyItem:any, i:number) {
+            const {label, count, percent} = data.datasets[tooltipModel.dataPoints[i].datasetIndex];
+            return {label, count, percent};
+        }
+
+        // Set Text
+        if (tooltipModel.body) {
+
+            const bodyLines:{ label:string; count:number; percent:number }[] = tooltipModel.body.map(getBody);
+            const {totalCases, altTotalPercent} = data.datasets[tooltipModel.dataPoints[0].datasetIndex];
+
+            let innerHtml = (
+                `<div><b>Summary for ${tooltipModel.title[0]}</b></div>
+                    <div>Gene altered in ${altTotalPercent}% of ${totalCases} cases</div>
+                    <table class="table table-striped">
+                        <thead>
+                            <tr>
+                                <th>Alteration</th>
+                                <th>Frequency</th>
+                            </tr>
+                        </thead>
+                        <tbody>`
+            );
+
+            bodyLines.forEach(body => {
+                innerHtml += (
+                    `<tr>
+                        <td> ${body.label} </td>
+                        <td> ${_.round(body.percent, 2)}% (${body.count} cases)</td>
+                     </tr>`
+                );
+            });
+            innerHtml += '</tbody></table>';
+
+            const tableRoot = tooltipEl.querySelector('div');
+            tableRoot!.innerHTML = innerHtml;
+        }
+
+        // `this` will be the overall tooltip
+        const position = chartOptions._chart.canvas.getBoundingClientRect();
+
+        // Display, position, and set styles for font
+        tooltipEl.style.opacity = '1';
+        // tooltipEl.style.display = 'block';
+        tooltipEl.style.left = position.left + tooltipModel.caretX + 'px';
+        tooltipEl.style.top = position.top + tooltipModel.caretY + 5 + 'px';
+    }
+
+    private componentDidMount() {
+        this.chartConfig.data = this.props.data;
+        this.chartConfig.options = this.chartOptions;
+        this.loadChart();
```

Comment:
because this code should only be called once, i would just put it up here rather than in method.